### PR TITLE
chore: add Gitpod support

### LINKED ISSUE #6: Add support for Gitpod
**Is your feature request related to a problem? Please describe.** A clear and
concise description of what the problem is. Ex. I'm always frustrated when [...]

Setting up an environment for a new contributor can be daunting. Give them an option where it's all set up for them.

**Describe the solution you'd like** A clear and concise description of what you
want to happen.

Add Gitpod support so they can work in the browser with a VS Code like experience

**Describe alternatives you've considered** A clear and concise description of
any alternative solutions or features you've considered.

Codespaces is another option, but not everyone has access to it.

**Additional context** Add any other context or screenshots about the feature
request here.

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [ ] Feature
- [ ] Bug Fix
- [ ] Documentation Update
- [ ] Refactor
- [ ] Optimization
- [x] Developer Experience

## Description

Adds support for working on the project in Gitpod.

TODO: Write docs related to this

## Related Tickets & Documents

- Related Issue #
- Closes #6

## QA Instructions, Screenshots, Recordings

1. To load GitPod for testing this PR, navigate to https://gitpod.io/#https://github.com/nickytonline/connect-four/tree/nickytonline/add-gitpod in your browser.

2. It'll take a minute to load up.

image

4. Gitpod will prompt to open in VS Code. Click the X in the top right of the
 prompt.

image

5. Gitpod will prompt you to install the Svelte for VS Code extension. Click the Install button.

image

7. It will install all npm packages for the project. Once all the dependencies are installed, it will load the project by running `ntl dev` and the app will appear in development mode.

image

8. From there, you can start developing like you would if working on your local machine's IDE.

### UI accessibility concerns?

N/A

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: Setting up a different environment for development
- [ ] I need help with writing tests

## [optional] What gif best describes this PR or how it makes you feel?

Chartacter from Southpark saying, "Nice!"

### CODE REVIEW

File: .gitpod.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM gitpod/workspace-full:latest
```
Comment (nickytonline) [inline .gitpod.Dockerfile:1]: Using a custom Docker config for Gitpod so we have the  latest security patched Node 16 version.

File: .gitpod.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+ports:
+  - port: 5173
+    onOpen: ignore
+  - port: 8888
+    onOpen: ignore
+
+tasks:
+  - name: 'Preview App'
+    command: gp ports await 5173 && gp preview $(gp url 5173)
```
Comment (nickytonline) [inline .gitpod.yml:15]: Thes are Gitpod commands. `gp` stands for Gitpod. For more information about `gp`, see the official [Gitpod CLI documentation](https://www.gitpod.io/docs/command-line-interface).

**Comment by nickytonline** [comment]
@Narigo, this is working, but I still need to update the documentation to mention Gitpod support.

**Comment by Narigo** [comment]
I've only set up docker compose for dev and prod environments in the past to make sure I have hot reload in my dev env and prod works out for deployment. Gitpod seems to set up an IDE as well, so the whole dev experience should be the same if you want to work on the project, do I understand that correctly?

In the PR, there is documentation about this - I suspect Gitpod only works fully once I've forked this project and used it as described there with the config file being merged already - is that correct? Would the documentation suffice then to set it up? I believe so, but if not, I would like to add an additional issue just for that so it won't be forgotten.

Btw, I'm not totally sure whether I like this approach - I have a few CLI tools and scripts that I regularly use and it looks like you would need to set them up for Gitpod in some way? Maybe if I put all my dev related tools into something that I can deploy / use in Gitpod it could make sense 🤔

Coming back to the review request: Please merge it, then we can actually test drive it and then let's see whether it works 👍

**Review by Narigo — APPROVED** [review]
Sorry, I thought I had used Review for my comments before! Go ahead! 👍

**Pull request merged**